sequence/model/lstm.py

```diff
@@ -0,0 +1,108 @@
+import torch
+import torch.nn.functional as F
+
+from sequence.model.modular import Embedding
+
+from torch import nn, optim
+
+import pickle
+import numpy as np
+
+
+class LSTM(Embedding):
+    def __init__(self, vocabulary_size, embedding_dim=10, custom_embeddings=None):
+        self.lstm_layers = 1
+        self.latent_size = 20
+        super().__init__(vocabulary_size, embedding_dim, custom_embeddings)
+
+        self.lstm = nn.LSTM(
+            input_size=embedding_dim,
+            hidden_size=self.latent_size,
+            num_layers=self.lstm_layers,
+            # dropout=0.2,
+            batch_first=True,
+        )
+        self.fc = nn.Linear(self.latent_size, vocabulary_size)
+
+    def forward(self, x, prev_state):
+        emb = self.apply_emb(x)
+        padded_emb, lengths = nn.utils.rnn.pad_packed_sequence(
+            emb, padding_value=0, batch_first=True
+        )
+        output, state = self.lstm(padded_emb, prev_state)
+        fc = self.fc(output)
+
+        return torch.log_softmax(fc, -1), state
+
+    def init_state(self, batch):
+        # TODO use random number for init
```

Comment:
Why random numbers? During the start of a sequence a nullified hidden state represent that there isn't any hidden state yet.